compressor/contrib/jinja2ext.py

```diff
@@ -45,11 +45,10 @@ def _compress_normal(self, kind, mode, caller):
 
     def _compress(self, kind, mode, caller, forced):
         mode = mode or OUTPUT_FILE
-        original_content = caller()
         context = {
-            'original_content': original_content
+            'original_content_generator': caller,
         }
         return self.render_compressed(context, kind, mode, forced=forced)
 
     def get_original_content(self, context):
-        return context['original_content']
+        return context['original_content_generator']()
```

Comment:
i don't know :) but @diox might...? if you can write a test that demonstrates the problem, i could merge this, but i'd also like to have this in a separate commit.